src/Model.php

```diff
@@ -613,26 +616,49 @@ public function getFields($filter = null): array
     {
         if ($filter === null) {
             return $this->fields;
-        } elseif (is_string($filter)) {
+        } elseif (!is_array($filter)) {
             $filter = [$filter];
         }
 
-        return array_filter($this->fields, function (Field $field, $name) use ($filter) {
-            // do not return fields outside of "only_fields" scope
-            if ($this->only_fields && !in_array($name, $this->only_fields, true)) {
-                return false;
-            }
-            foreach ($filter as $f) {
-                if (
-                    ($f === 'system' && $field->system)
-                    || ($f === 'not system' && !$field->system)
-                    || ($f === 'editable' && $field->isEditable())
-                    || ($f === 'visible' && $field->isVisible())
-                ) {
+        $filterPresets = [
```

Comment:
good, but declare inside with if directly - declaring array again and again is too costly for this core thing

Reply:
@georgehristov ping

Reply:
Not sure what you mean.
We accept string, array of strings (also closures)
And we return matching with OR

Reply:
In current design you create `$filterPresets` array with callbacks every time `getFields()` is called.

I think we should no create array like this every time and check like before directly.

Reply:
Better now?